doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,605 @@
+
+Example of a list of connection points:
+
+::
+
+    - ingress:
+          type: interface
+          interface: 2
+      egress:
+          type: interface
+          interface: 1
+      mpls_label: 1    
+    - ingress:
+          type: interface
+          interface: 2
+      egress:
+          type: ipv4
+          ip: 192.168.5.1
+          prefix: 24
+      mpls_label: 55
+    - ingress:
+          type: wildcard
+      egress:
+          type: interface
+          interface: 2
+      mpls_label: 3
+
+
+
+Border router
+^^^^^^^^^^^^^^^
+
+For a router to query the DRKey secret value from the control service, once has to enable this.
+
+This could look like this::
+
+    [router]
+    use_drkey = true
+
+Considerations for future work
+--------------------------------
+
+SCMP response
+^^^^^^^^^^^^^^^
+
+With the current implementation, the sending endhost is not being informed when his packet gets dropped due to a FABRID error.
+In the future the border routers we might think about whether to enable routers to send an SCMP response if they encounter
+an error when processing FABRID which might help the sending endhost in figuring out why his packet does not arrive at its destination.
+However, we have to be careful to not introduce additional attack vectors, e.g. for volumetric denial-of-service or reflection attacks.
+
+EPIC-HP as extension with Identifier option
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We could create a new HBH extension for EPIC hidden-path, which uses the Identifier option, instead of the
+current design which uses a SCION path-type.
+This allows the use of EPIC HP also in a incremental deployment like we have with FABRID.
+And additionally, we could also use FABRID together with EPIC HP.
+
+.. _RAINBOW:
+
+The RAINBOW system
+^^^^^^^^^^^^^^^^^^^^^
+
+The RAINBOW system allows marking traffic as higher quality of service, to be
+prioritized at on-path BRs.
+RAINBOW can be implemented based on FABRID.
+Specifically, each border router reserves a certain amount of bandwidth for RAINBOW traffic, which is divided evenly between different source ASes.
+As this division happens on the level of ASes and not individual endhosts, the BR also needs to be able to authenticate traffic at an AS level.
+The FABRID HBH extension makes this possible, by including an "AS level key" flag, which specifies that the original HVF has been authenticated
+again by an AS-AS DRKey, that is only known to trusted infrastructure in the source AS.
```

Comment:
by the corresponding AS-AS DRKey, which is only

Reply:
Done